components/TransactionEntryForm/index.js

```diff
@@ -29,141 +25,130 @@ function TransactionEntryForm({
+          />
+        </Styled.FormField>
+        <Styled.FormField>
+          <label htmlFor="amount">Amount</label>
+          <input
+            type="number"
+            name="amount"
+            max="999999.99"
+            min="0.00"
+            step="0.01"
+            required
+            defaultValue={mode === "add" ? "" : currentTransaction.amount}
+          />
+        </Styled.FormField>
+        <Styled.FormField>
+          <label htmlFor="currency">Currency</label>
+          <select
+            type="text"
+            name="currency"
+            required
+            defaultValue={
+              mode === "add" ? "Default" : currentTransaction.currency
+            }
+          >
+            <option value="Default">--Select Option--</option>
+            <option value="€">€</option>
+          </select>
+        </Styled.FormField>
+        <Styled.FormField>
+          <label htmlFor="date">Date</label>
+          <input
+            type="date"
+            name="date"
+            required
+            defaultValue={mode === "add" ? "" : currentTransaction.date}
+          />
+        </Styled.FormField>
+        <Styled.FormField>
+          <label htmlFor="paymentMethod">Payment Method</label>
+          <select
+            type="text"
+            name="paymentMethod"
+            required
+            defaultValue={
+              mode === "add" ? "" : currentTransaction.paymentMethod
+            }
+          >
+            <option value="Default">--Select Option--</option>
+            <option value="Cash">Cash</option>
+            <option value="Card">Card</option>
+          </select>
+        </Styled.FormField>
+        <Styled.FormField>
+          <label htmlFor="category">Category</label>
+          <select
+            type="text"
+            name="category"
+            required
+            defaultValue={selectedCategory}
+            onChange={(event) => setSelectedCategory(event.target.value)}
+          >
+            <option value="Groceries">Groceries</option>
+            <option value="Salary">Salary</option>
+            <option value="Housing">Housing</option>
+            <option value="Insurance">Insurance</option>
+            <option value="Utilities">Utilities</option>
+          </select>
+        </Styled.FormField>
+        <Styled.FormField>
+          <label htmlFor="description">Description</label>
+          <textarea
+            rows="7"
+            maxLength={140}
+            type="text"
+            name="description"
+            defaultValue={mode === "add" ? "" : currentTransaction.description}
+          />
+        </Styled.FormField>
+      </Styled.FormContainer>
+      <Styled.FormButton>
+        <Button $type="submit">{mode === "add" ? "Add" : "Save"}</Button>
+        <Button
+          $visible={mode === "edit"}
           $type="danger"
           $textColor="white"
           onClick={() => setShowModal(false)}
-          >Cancel</Button>
```

Comment:
is there a specific reason why you use css to hide the button instead of a ternary operator or logic operator?
Counterexample: 
```jsx
{
  mode === "edit" && <Button>...</Button>
}
```
This hides the button as well. Both version are fine, but you should choose one variant and stick with it throughout the project. This makes the code predictable and easier to understand.